tests/test_tally.py

```diff
@@ -0,0 +1,213 @@
+from pathlib import Path
+from typing import Sequence, Type
+
+import build123d as bd
+import numpy as np
+import openmc
+import openmc.stats
+import pytest
+import stellarmesh as sm
+
+RELATIVE_TOLERANCE_PERCENT = 2.5 / 100
+
+logger = logging.getLogger(__name__)
+
+
+class Model(metaclass=ABCMeta):
+    @cached_property
+    @abstractmethod
+    def materials(self) -> openmc.Materials:
+        """Generate the neutron source."""
+        ...
+
+    @abstractmethod
+    def tallies(self) -> openmc.Tallies:
+        """Generate the tallies."""
+        ...
+
+    @abstractmethod
+    def source(self) -> openmc.IndependentSource:
+        """Generate the neutron source."""
+        ...
+
+    @abstractmethod
+    def csg(self) -> openmc.Model:
+        """Generate the CSG model."""
+        ...
+
+    @abstractmethod
+    def dagmc(self) -> sm.DAGMCModel:
+        """Generate the DAGMC model."""
+        ...
+
+    def _make_dagmc_model(
+        self,
+        solids: Sequence[bd.Solid],
+        materials: openmc.Materials,
+        min_mesh_size: float,
+        max_mesh_size,
+    ) -> sm.DAGMCModel:
+        mat_names: list[str] = [m.name for m in self.materials]
+        geom = sm.Geometry(solids, mat_names)
+        msh = sm.Mesh.from_geometry(geom, min_mesh_size, max_mesh_size)
+        return sm.DAGMCModel.from_mesh(msh)
+
+
+class NestedSpheres(Model):
+    """Two nested spheres."""
+
+    radius1 = 20
+    radius2 = 5
+
+    @cached_property
+    def materials(self):
+        mat1 = openmc.Material(name="1")
+        mat1.add_nuclide("Fe56", 1)
+        mat1.set_density("g/cm3", 1)
+
+        mat2 = openmc.Material(name="2")
+        mat2.add_nuclide("Be9", 1)
+        mat2.set_density("g/cm3", 1)
+        return openmc.Materials([mat1, mat2])
+
+    def tallies(self):
+        mat1_filter = openmc.MaterialFilter(self.materials[0])
+        tally1 = openmc.Tally(name="mat1_flux_tally")
+        tally1.filters = [mat1_filter]
+        tally1.scores = ["flux"]
+
+        mat2_filter = openmc.MaterialFilter(self.materials[1])
+        tally2 = openmc.Tally(name="mat2_flux_tally")
+        tally2.filters = [mat2_filter]
+        tally2.scores = ["flux"]
+
+        return openmc.Tallies([tally1, tally2])
+
+    def source(self):
+        source = openmc.IndependentSource()
+        source.space = openmc.stats.Point((0, 0, 0))
+        source.angle = openmc.stats.Isotropic()
+        source.energy = openmc.stats.Discrete([14e6], [1])
+        return source
+
+    def csg(self):
+        surface1 = openmc.Sphere(r=self.radius1)
+        surface2 = openmc.Sphere(r=self.radius1 + self.radius2, boundary_type="vacuum")
+        region1 = -surface1
+        region2 = +surface1 & -surface2
+        cell1 = openmc.Cell(region=region1)
+        cell1.fill = self.materials[0]
+        cell2 = openmc.Cell(region=region2)
+        cell2.fill = self.materials[1]
+        csg_geometry = openmc.Geometry(openmc.Universe(cells=[cell1, cell2]))
```

Comment:
Common shorthand that I use:
```suggestion
        csg_geometry = openmc.Geometry([cell1, cell2])
```

Reply:
Agreed this is better, but just fails type checking because of that bug I reported. Will do anyways